internal/testutils/restartable_listener.go

```diff
@@ -83,11 +83,10 @@ func (l *RestartableListener) Addr() net.Addr {
 func (l *RestartableListener) Stop() {
 	l.mu.Lock()
 	l.stopped = true
-	tmp := l.conns
-	l.conns = nil
-	for _, conn := range tmp {
+	for _, conn := range l.conns {
```

Comment:
Comment?  `// Close conns with the lock held because ________`

Reply:
I remember I added this because one of the tests was not happy with the race detector enabled without this change. But now, I am not able to reproduce that failure. I tried running all tests a few times, but all of the them seem to pass. I can revert this change if you feel this is not required.

Reply:
I don't think there's any need to revert it.  No need for a comment here, then, either, since if we did want to make this change in the future, we apparently can.